Jenkinsfile

```diff
@@ -416,6 +418,21 @@ lock(resource: "build-${params.STREAM}") {
                     """)
                 }
             }
+
+            // If there is a config for GCP then we'll upload our image to GCP
+            if (utils.path_exists("\${GCP_IMAGE_UPLOAD_CONFIG}")) {
```

Comment:
Can we run this in parallel with the `Upload AWS` stage?

Reply:
yeah - I started to copy over some paralllel code and doing it that way to begin with but both of them update the metadata IIUC and I wanted to just sprint to getting this in and being able to do followups later.

Reply:
Ahhh right, that issue. OK yup, let's leave that for later. An easy fix for this would be to add locking in cosa for the crucial bits.